### PR TITLE
B2B-1747: [NEW] account hierarchy endpoints

### PR DESCRIPTION
**Pull request by JamessenLiu**
# [B2B-1747](https://bigcommercecloud.atlassian.net/browse/B2B-1747)

## What changed?

* added parent companies fieds in company list and detail resposne
* new endpoint: **get subsidiaries**
* new endpoint: **delete subsidiary**
* new endpoint: **attach parent company**
* new endpoint: **get hierarchy**

## Release notes draft

n/a

## Anything else?

n/a

[B2B-1747]: https://bigcommercecloud.atlassian.net/browse/B2B-1747?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 1]
```diff
@@ -3052,7 +3081,452 @@ paths:
...
+                        examples:
+                          parameter error:
+                            value:
+                              code: 400
+                              data: {
+                                errMsg: Invalid child company ID.
+                              }
+                              meta:
+                                message: Parameter error.
+  "/companies/{companyId}/parent":
```
[the remark this answers is not in the document]
Comment (bc-traciporter) [reply docs/b2b-edition/specs/api-v3/company/company.yaml:3259]: Add the `companyId` query parameter at the path level for the following endpoints.  The ref callout is not working.

- /companies/{companyId}/catalogs
- /companies/{companyId}/status:

parameters:
        - name: companyId
          in: path
          required: true
          schema:
            type: string

Or you can fix the alignment of the components > parameters > companyId.
Move components all the way to the left side.

File: docs/b2b-edition/specs/api-v3/company/company.yaml [hunk 2]
```diff
@@ -3052,7 +3081,452 @@ paths:
...
+                                    companyStatus:
+                                      type: integer
+                                      description: 'The status of the company. 0: PENDING 1: APPROVED 2: REJECTED 3: INACTIVE.'
+                                      enum:
+                                        - 0
+                                        - 1
+                                        - 2
+                                        - 3
+                                    subsidiaries:
+                                      type: array
```
Comment (bc-traciporter) [inline docs/b2b-edition/specs/api-v3/company/company.yaml:3443]: This array requires an items sibling.
Is this structure correct? subsidiaries within subsidiaries?

**Review by bc-traciporter — COMMENTED** [review]
Left comments that caused errors in the file.

**Comment by bc-traciporter** [comment]
Once these are fixed, I will start moving the contents over to the private repo.

File: docs/b2b-edition/specs/api-v3/company/company.yaml (hunk 2 above)
Comment (JamessenLiu) [reply docs/b2b-edition/specs/api-v3/company/company.yaml]: @bc-traciporter 
Yes, subsidiaries can also have subsidiaries. This is a schematic diagram of the company hierarchy.
image

**Comment by JamessenLiu** [comment]
@bc-traciporter 
Please review again, thinks.

**Comment by bc-traciporter** [comment]
Hi @JamessenLiu The PR looks good. I will move this over to the private repo.

**Comment by JamessenLiu** [comment]
@bc-traciporter 
> I don't understand the fieldName description. "Field name that config in you store by the B3 team". Who is the B3 team? Can I change it to "The B3 team configures the field name in your store."

The B3 team is an abbreviation for the former B2B team. It is inappropriate to continue using the term B3 team here. Extra fields are now fully available to merchants, and there is no longer a need for the B2B team to assist merchants with configuration. I have updated the field descriptions in that section.

> The description for priceListAssign is "This field is only enable for MSF store". I would like to change this to "You can only enable this field for an MSF store."

updated

> What does "The API rate limit is 15/min per store." mean? Should it say "The API rate limit is 15 per store".

What I want to express is that each store can only call this API 15 times per minute. I would like to change the description to: "The API rate limit is 15 requests per store per minute." Is that appropriate?

**Comment by bc-traciporter** [comment]
Thank you @JamessenLiu I added these changes. Do you plan to include changes for [B2B-1761](https://bigcommercecloud.atlassian.net/browse/B2B-1761) to this PR or is this everything?

**Comment by JamessenLiu** [comment]
Hi, @bc-traciporter 
Currently, all documents related to the company hierarchy are included in this PR, including [B2B-1761](https://bigcommercecloud.atlassian.net/browse/B2B-1761).

**Comment by bc-traciporter** [comment]
Thank you @JamessenLiu. I will close this PR. I will finish the beta doc and give you permissions to view it. I will use jamessen.liu@bigcommerce.com for your email address.

**Comment by bc-traciporter** [comment]
Hi @JamessenLiu I am still in the process of getting the beta doc ready and I was contacted about this file. I had merged it  into the api-spec-ssot repo and was told it caused an error that prevented a SDK from compiling. So I removed `minLength: 1` from companyStatus to correct the issue. companyStatus is listed as a required field. Please let me know if you have any issues.

image

**Pull request closed without merging**